src/group/mls_group/new_from_welcome.rs

```diff
@@ -59,7 +59,13 @@ impl MlsGroup {
                 return Err(WelcomeError::DuplicateRatchetTreeExtension);
             }
             match extension.to_ratchet_tree_extension_ref() {
-                Ok(e) => Some(e.clone()),
+                Ok(e) => {
```

Comment:
I only expected changes in the test. What is this about?
If this is needed, the extension shouldn't be removed in the first place.

Reply:
Ah, good point. I explained it in the commit message, but I should have added it in the PR description as well.

Yeah, I didn't really know how to solve this nicely. The existing code checks for duplicates by removing the existing extension and then checking if there are still `any` left. So I had the option of either implementing a new deduplication strategy or just sticking with what is already there and adding it back afterwards...

Reply:
Sorry, I guess you should have looked at the commit message a little closer 😬 (I really don't like GH reviews...)
Ok let's keep it like this for now. This looks like something that could use a cleanup in any case later.

Reply:
No worries. I should have changed the PR description or at least made a comment in the thread.